text/0308-cli-advisories.md

```diff
@@ -0,0 +1,263 @@
+advisories.
+
+### What future improvements can we make?
+
+**Resource filter**: Ideally, advisories should be as targeted as possible. The
+version filter addresses part of that requirement, but we can make it better by
+also adding a resource filter: if an issue only affects `FargateTaskDefinition`
+resources, for example, but this resource is not present in any of the stacks,
+the customer doesn’t need to see that particular warning.
+
+**Pipeline awareness**: Initially, we won’t treat the pipeline as a special
+case. By default, the advisories will be fetched and displayed and we expect
+users to never look at them. This behavior can always be overridden by
+suppressing the advisories. In future releases, the CLI will use the information
+about whether it’s running on a pipeline and block the deployment in case there
+is any advisory.
+
+## Appendix
+
+### Detailed design
+
+There are three parts to this solution: the data format, the GitHub workflow and
+the CLI logic.
+
+#### Data format
+
+We will use GitHub issues to publish and manage advisories. To be considered a
+valid advisory, the body of a GitHub issue must have the following fields:
+
+| Field                  | Description                                                                                                                                                | Format                                 | Mandatory? |
+| ---------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------- | ---------- |
+| Summary                | A paragraph with more information                                                                                                                          | Free-form text                         | Yes        |
+| Affected component     | The CLI or the construct library (or both)                                                                                                                 | A non-empty subset of {CLI, Framework} | Yes        |
+| Affected version range | Version range using the semver format                                                                                                                      | semver                                 | No         |
+| Suggested fix          | An action the user can take to mitigate the problem. If it requires a longer explanation, this field should be omitted in favor of the resource linked to. | Free-form text                         | No         |
+
+In order to keep the issue human readable as well as easily parseable, each of
+these fields should be indicated by a Markdown header (`### Summary`, `###
+Affected component` etc). The piece of text following the header will be
+interpreted as the content of the field.
+
+#### GitHub workflow
+
+To be considered an advisory, a GitHub issue must be pinned and have the
```

Comment:
I am still wondering if from a workflow perspective it makes more sense to publish this by committing a JSON file in the aws-cdk's main branch. This will have the very important benefit of integrating with our PR workflow and will allow us to run validation checks and human approvals on these advisories before they are published.

If we want, we can easily create a GitHub workflow that will automatically create GitHub issues based on this JSON file so advisories will _also_ appear as pinned GitHub issues, but the source of truth will be the source repo and no the issues.

@rix0rrr @otaviomacedo what do you think?

Reply:
That's definitely an option. Let me think about the trade-offs involved.

Reply:
@eladb the file wouldn't live only in `master`, but also in `v2-main`. Now that v2 is GA, both branches are reasonable candidates for the source of truth. Is it acceptable to just pick one?

Reply:
I think a single file on `master` or even in a dedicated repository is preferable. I am not sure I see strong coupling to major version that justifies multiple sources of truth.

Reply:
I'm still more inclined to authoring the issue directly. It already has a natural identifier and a URL for more information. And, in a way, the issue itself is also a source of truth for users, where they can see status updates, comments, other issues that linked to it etc. So we might as well embrace that and make it the single source of truth.

Reply:
OK, but I think we need some kind of an approval and verification workflow for these advisories. 

We need to verify that: (a) that the issue is formatted exactly based on what the client expects and then we need some monitoring if the advisory issue is "broken"; and (b) that there's some approval process for publishing an advisory, the content is reviewed etc.